### DISCUSSION TITLE
Is it possible to receive eth_sendTransaction and other "standard" rpc requests?

### DISCUSSION THREAD

**Discussion by chntr** [MetaMask/snaps#692] [Q&A] [answered]
Title says it all. 

I want my snap to check what was sent to eth_sendTransaction (does not need to modify). 

Thanks!

**Comment by ritave** [comment, marked as answer]
Hey @chntr, we're currently expanding what snaps can do with [Transaction Insight API](https://github.com/MetaMask/snaps-skunkworks/pull/642).

Your snap will be able to get information about transactions sent by the user.

```typescript
exports.onTransaction = ({ origin, transaction }) => console.log("Transaction data", transaction);
```

**Reply by chntr** [reply]
Thanks @ritave ! Will follow that PR

**Comment by schmanu** [comment]
Hey @ritave,

Any plans to extend this by an `onSignature`  Export / Signature Insight API?

I'm working on a offchain signing integration for smart contract wallets and would like to explore using MetaMask snaps to show insights for a signature.

**Reply by ritave** [reply]
[Yes](https://github.com/MetaMask/snaps-monorepo/issues/1024) :)
